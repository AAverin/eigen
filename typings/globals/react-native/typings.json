{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
