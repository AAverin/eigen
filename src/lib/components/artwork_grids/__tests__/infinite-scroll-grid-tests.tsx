import 'react-native'

import * as React from 'react'
import { renderWithLayout } from '../../../tests/render_with_layout'

import InfiniteScrollArtworksGrid from '../infinite_scroll_grid'

it('renders properly', () => {
  const artist = {
    artworks: {
      edges: [ artwork(), artwork(), artwork() ]
    }
  }

  const grid = renderWithLayout(<InfiniteScrollArtworksGrid artist={artist} queryKey="artist" />, { width: 768})
  expect(grid).toMatchSnapshot()
})

var artwork = () => {
  return {
    node: {
      __id: Math.random(),
      title: 'TALKING POLITICS',
      date: '2008',
      sale_message: 'Not for sale',
      is_in_auction: false,
      image: {
        url: 'artsy.net/image-url',
        aspect_ratio: 1.27
      },
      artists: [
        {
          name: 'Jenny Holzer'
        }
      ],
      partner: {
        name: 'Kukje Gallery'
      },
      href: '/artwork/jenny-holzer-talking-politics'
    }
  }
}
