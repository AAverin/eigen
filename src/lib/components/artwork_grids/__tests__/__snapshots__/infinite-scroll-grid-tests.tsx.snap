exports[`test renders properly 1`] = `
<ArtworksGrid
  artist={
    Object {
      "artworks": Object {
        "edges": Array [
          Object {
            "node": Object {
              "__id": 0.16210295153931287,
              "artists": Array [
                Object {
                  "name": "Jenny Holzer",
                },
              ],
              "date": "2008",
              "href": "/artwork/jenny-holzer-talking-politics",
              "image": Object {
                "aspect_ratio": 1.27,
                "url": "artsy.net/image-url",
              },
              "is_in_auction": false,
              "partner": Object {
                "name": "Kukje Gallery",
              },
              "sale_message": "Not for sale",
              "title": "TALKING POLITICS",
            },
          },
          Object {
            "node": Object {
              "__id": 0.28603492766972405,
              "artists": Array [
                Object {
                  "name": "Jenny Holzer",
                },
              ],
              "date": "2008",
              "href": "/artwork/jenny-holzer-talking-politics",
              "image": Object {
                "aspect_ratio": 1.27,
                "url": "artsy.net/image-url",
              },
              "is_in_auction": false,
              "partner": Object {
                "name": "Kukje Gallery",
              },
              "sale_message": "Not for sale",
              "title": "TALKING POLITICS",
            },
          },
          Object {
            "node": Object {
              "__id": 0.24393106130356568,
              "artists": Array [
                Object {
                  "name": "Jenny Holzer",
                },
              ],
              "date": "2008",
              "href": "/artwork/jenny-holzer-talking-politics",
              "image": Object {
                "aspect_ratio": 1.27,
                "url": "artsy.net/image-url",
              },
              "is_in_auction": false,
              "partner": Object {
                "name": "Kukje Gallery",
              },
              "sale_message": "Not for sale",
              "title": "TALKING POLITICS",
            },
          },
        ],
      },
    }
  }
  queryKey="artist" />
`;
